great_expectations/datasource/datasource_serializer.py

```diff
@@ -0,0 +1,83 @@
+"""Serialize Datasource Configurations.
+
+Serializers determine how to write an object to disk, json, etc.
+A serializer comprises the object destination and name e.g. YAMLReadyDictDatasourceConfigSerializer.
+
+Typical usage example:
+
+datasource_config = DatasourceConfig(...)
+serializer = YAMLReadyDictDatasourceConfigSerializer()
+serialized_value = serializer.serialize(datasource_config)
+"""
+from typing import Optional
+
+from great_expectations.core.serializer import AbstractConfigSerializer
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.data_context.types.base import (
+    DatasourceConfig,
+    datasourceConfigSchema,
+)
+from great_expectations.marshmallow__shade import Schema
+
+
+class YAMLReadyDictDatasourceConfigSerializer(AbstractConfigSerializer):
+    def __init__(self, schema: Optional[Schema] = None) -> None:
```

Comment:
We always want `datasourceConfigSchema` here right? Is overriding behavior accepted?

Reply:
No it's not - trying to think of a better pattern here.

Reply:
Trying not to have different signatures for parent/child